sympy/calculus/util.py

```diff
@@ -622,7 +622,7 @@ def lcim(numbers):
 
     return result
 
-def is_convex(f, *syms, **kwargs):
+def is_convex(f, *syms, domain=S.Reals):
```

Comment:
There have been issues in the past where having S objects at the module level forces it to create the instance at import time, which slows down import time at best and makes things hard to debug at worst. Hopefully this doesn't cause too many problems like that. IMO shoving all these things on S was a bad idea.